sound/soc/sof/trace.c

```diff
@@ -31,23 +30,37 @@ static size_t sof_wait_trace_avail(struct snd_sof_dev *sdev,
 	if (host_offset > pos)
 		return host_offset - pos;
 
+	return 0;
+}
+
+static size_t sof_wait_trace_avail(struct snd_sof_dev *sdev,
+				   loff_t pos, size_t buffer_size)
+{
+	wait_queue_entry_t wait;
+	size_t ret = sof_trace_avail(sdev, pos, buffer_size);
+
+	/* data immediately available */
+	if (ret)
+		return ret;
+
+	/* tracing has ended, signal EOF to client */
+	if (!sdev->dtrace_is_enabled && sdev->dftrace_eof) {
+		sdev->dftrace_eof = 0;
+		return 0;
+	}
+
```

Comment:
I would first split this function in two in a separate patch, then add the new functionality and add the renames last if needed. It's really confusing to review as is.